src/video/SDL_pixels.c

```diff
@@ -1235,106 +1235,135 @@ void SDL_DetectPalette(SDL_Palette *pal, SDL_bool *is_opaque, SDL_bool *has_alph
 }
 
 /* Find the opaque pixel value corresponding to an RGB triple */
-Uint32 SDL_MapRGB(const SDL_PixelFormat *format, Uint8 r, Uint8 g, Uint8 b)
+Uint32 SDL_MapRGB(SDL_PixelFormat format, SDL_Palette *palette, Uint8 r, Uint8 g, Uint8 b)
 {
-    if (!format) {
+    if (palette) {
```

Comment:
May be worth separating out the format vs palette functions, since they are separate code paths anyway?